module/module.go

```diff
@@ -467,6 +470,13 @@ func (m *Module) AddResource(ctx context.Context, req *pb.AddResourceRequest) (*
 		return nil, err
 	}
 
+	// If context has errored, even if construction succeeded we should close the resource and return the context error
+	// Use the shutdown context because otherwise any Close operations that rely on the context will immediately fail
+	if ctx.Err() != nil {
```

Comment:
What deadline does this context carry? Do we have the plumbing set up to get the 1 minute timeout from the RDK on this context? I'm assuming logic like this might not be possible in C++ and Python?

Reply:
The deadline should be one minute, will update the comment to reference
from RDK to Python SDK, the plumbing exists (most likely for the C++ SDK as well). The deadline is carried over grpc, and should be available to the method.

Reply:
added